tests/acceptance/dns_test.go

```diff
@@ -203,36 +174,22 @@ var _ = Describe("DNS", func() {
 			return records[0].String(), nil
 		}).Should(Equal("1.2.3.5"))
 
-		By("creating an A record for the ingress")
-		Eventually(func() error {
-			var err error
-			records, err = resolver.LookupIP(ctx, "ip", ingressDomain)
-			return err
-		}).Should(Succeed())
-
-		Expect(records).To(HaveLen(1))
-		Expect(records[0].String()).To(Equal("10.0.0.2"))
-
 		By("creating a CNAME record for the wildcard domain")
 		wildcardDomain := fmt.Sprintf("%s.%s", uuid.NewString(), clusterDomain)
-		var record string
+		var dnsResponse *dns.Msg
 		Eventually(func() error {
 			var err error
-			record, err = resolver.LookupCNAME(ctx, wildcardDomain)
-			return err
-		}).Should(Succeed())
+			dnsMessage := new(dns.Msg)
+			dnsMessage.SetQuestion(wildcardDomain, dns.TypeCNAME)
+			dnsMessage.RecursionDesired = true
 
-		Expect(record).To(Equal(ingressDomain))
+			dnsClient := new(dns.Client)
+			dnsResponse, _, err = dnsClient.Exchange(dnsMessage, "8.8.8.8:53")
 
-		By("creating an A record for the wildcard domain")
-		var hostRecords []string
-		Eventually(func() error {
-			var err error
-			hostRecords, err = resolver.LookupHost(ctx, wildcardDomain)
 			return err
 		}).Should(Succeed())
 
-		Expect(hostRecords).To(ConsistOf("10.0.0.2"))
```

Comment:
And during tests we don't have `external-dns`, so we should not test that the A record behind it's correct.

Reply:
I think we should use this client for all the other tests too, just to be consistent. It is however very ugly, so I don't know how I feel about it. Maybe we can also hide how the message is constructed and the answer extracted with some functions.

Also there's no need to construct the message and client in the eventually func.